### DISCUSSION TITLE
...src/libvalhalla.a(options.pb.cc.o): undefined reference to symbol... when compiling Valhalla 3.4.0

### DISCUSSION THREAD

**Discussion by LZhaoM** [valhalla/valhalla#4711] [Q&A] [answered]
Hi,
When I try to compile Valhalla 3.4.0 from source on a linux system, I get the following issues:
```
[ 61%] Built target valhalla-tyr
[ 61%] Built target valhalla
[ 61%] Linking CXX executable valhalla_run_map_match
/usr/bin/ld: src/libvalhalla.a(options.pb.cc.o): undefined reference to symbol '_ZN4absl12lts_2024011612log_internal10LogMessage19CopyToEncodedBufferILNS2_10StringTypeE0EEEvSt17basic_string_viewIcSt11char_traitsIcEE'
/usr/bin/ld: /usr/lib/libabsl_log_internal_message.so.2401.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/valhalla_run_map_match.dir/build.make:106: valhalla_run_map_match] Error 1
make[1]: *** [CMakeFiles/Makefile2:1148: CMakeFiles/valhalla_run_map_match.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```
I didn't run `install-linux-deps.sh` because the system is arch linux.
And I am trying to make some changes based on Valhalla 3.4.0, so I want to build it from source(more convenient for developing). Or it is a better idea to make changes based on the latest commit?
Thank you for any help!

**Comment by LZhaoM** [comment, marked as answer]
I just realised I just need to deploy it on an Ubuntu environment. Please forget about it :).

**Reply by nilsnolde** [reply]
It doesn’t need to be Ubuntu, I’m using Manjaro (basically Arch). But yeah, always use latest master
